onadata/libs/tests/utils/test_export_tools.py

```diff
@@ -349,6 +401,50 @@ def test_should_not_create_new_export_when_old_exists(self):
             Export.objects.get(xform=self.xform).options,
         )
 
+        # New metadata will yield a new export
+        metadata.delete()
```

Comment:
@FrankApiyo Will updating `metadata.extra_data` suffice instead of deleting and creating a new one?

Reply:
Whenever you create a new linked dataset, a new metadata object is created; So creating a new metadata object more closely simulates what happens in production. (as compared to editing an existing metadata object)

Reply:
@FrankApiyo Noted. Though l believe since this is a unit test and not an integration test, we are only interested in the `metadata` passed to `custom_response_handler` and not creation of the linked dataset.